frontend/packages/console-plugin-sdk/src/typings/storage-class-params.ts

```diff
@@ -0,0 +1,16 @@
+import { Extension } from './base';
+
+namespace ExtensionProperties {
+  export interface StorageClassProviosner {
+    getStorageClassProvisoner: () => any;
+  }
+}
+
+export interface StorageClassProviosner
+  extends Extension<ExtensionProperties.StorageClassProviosner> {
+  type: 'StorageClass/Proviosner';
+}
+
+export function isStorageClassProvisioner(e: Extension): e is StorageClassProviosner {
+  return e.type === 'StorageClass/Proviosner';
+}
```

Comment:
This is specific to OCS hence we can keep it in ceph package. If required in future we can move to shared.
We already have one extensions folder under /src

Reply:
Can we contribute type definitions to SDK from plugins?

Reply:
@afreen23 imo, we shouldnt contribute type definations to sdk from plugins if they are to be used in public component.

Reply:
I missed this PR. Sorry for late response.
Many plugins have their extensions in their own package. We have one for ceph too already (which is stale now).
I think ceph csi is the consumer for this extension hence moving to ceph package makes sense to me.
https://github.com/openshift/console/blob/dccfe8fc07cbdfadacfa2d9b11f5fab8e3d71192/frontend/packages/ceph-storage-plugin/src/extensions/dashboards.ts#L1-L12

Reply:
@afreen23  Im not sure I follow. The current place is ok. `ceph-storage-plugin/src/extensions/*.ts` is for other plugins that want to extend `ceph-storage-plugin`. In this case we are extending core so the extension definition should live in `console-plugin-sdk`.

Reply:
@afreen23 the plugin is generic, it can be used both by csi and non-csi provisioners. It can help put custom components to any provisioner. imo, Its better to keep it in SDK for its better usage.

Reply:
> @afreen23 Im not sure I follow. The current place is ok. `ceph-storage-plugin/src/extensions/*.ts` is for other plugins that want to extend `ceph-storage-plugin`. In this case we are extending core so the extension definition should live in `console-plugin-sdk`.

Agree, seems like I was suggesting to derail from current patterns. My bad !

Reply:
> @afreen23 the plugin is generic, it can be used both by csi and non-csi provisioners. It can help put custom components to any provisioner. imo, Its better to keep it in SDK for its better usage.

Sure @a2batic :+1: